.gitignore

```diff
@@ -23,6 +23,7 @@ build
 .tox
```

Comment:
@AdrianSosic can you elaborate on this `and the attribute doesn't refer to a callable but really to a transform "mode".` ? They are quite literally specifying which `function` should be used

I agree with the bounds part of the arg, but calling it mode resulting on two new modes that are potentially confusing to the users imo is not best. Now we have `TargetMode` and `TargetTransformMode`, is that really bette r than `TargetMode` and `TargetTransformFunction` ?

Reply:
I understand your point that `mode` and `transform_mode` is too similar and I would also wish for better ideas here. What I dislike a bit about using `transform_function` though is that it implies to me that I can pass the actual callable that carries out the transformation. But this is not the case; instead, we only pass a "mode" (or call it "type" if you prefer... perhaps that solves the issue, i.e. `transform_type`?) that is used as a selector for fixed choices of functions. What do you suggest?

Reply:
what about just `transform` or `transformation`? So a `Target` would have `mode` and `transformation`. Ultimately if you want to keep `transform_mode` its not a hardblock for me tbh

Reply:
No that's fine, will change it accordingly and ping you once done. Any other issues from your end? Would try to get this merge-ready as soon as possible :)

Reply:
Done. Had to use `transformation` because our action transformation method is already called `transform`. But for me everything's fine now. Please close if you approve